src/transformers/trainer.py

```diff
@@ -303,6 +322,12 @@ def __init__(
         # Mixed precision setup
         self.use_apex = False
         self.use_amp = False
+
+        # deepspeed manages its own fp16, so don't interfere
+        # XXX: perhaps assert if --fp16 is passed with --deepspeed?
+        if args.deepspeed:
+            args.fp16 = False
```

Comment:
I'd be in favor of a `raise ValueError(...)` here. Think it's better then silently setting it to False

Reply:
We should never change the `TrainingArguments` inside the `Trainer` as a general rule. A `ValueError` can be raised instead.

Reply:
Understood and Done. Thank you!